File: apps/api/app/schemas/forecast.py
from typing import Any, Literal

from pydantic import BaseModel, ConfigDict, Field


class GeoJSONFeature(BaseModel):
    model_config = ConfigDict(extra="forbid")

    type: Literal["Feature"] = "Feature"
    geometry: dict[str, Any]
    properties: dict[str, Any] = Field(default_factory=dict)


class GeoJSONFeatureCollection(BaseModel):
    model_config = ConfigDict(extra="forbid")

    type: Literal["FeatureCollection"] = "FeatureCollection"
    features: list[GeoJSONFeature]
    metadata: dict[str, Any] = Field(default_factory=dict)

File: apps/mobile/src/types/forecast.ts
export type GeoJSONFeature = {
  type: 'Feature';
  geometry: Record<string, unknown>;
  properties: Record<string, unknown>;
};

export type GeoJSONFeatureCollection = {
  type: 'FeatureCollection';
  features: GeoJSONFeature[];
  metadata: Record<string, unknown>;
};

export type ForecastLayerSummary = {
  totalFeatures: number;
  observations: number;
  knownRecords: number;
  possibleCorridors: number;
  samplingGaps: number;
  staticContext: number;
};
